index.html

```diff
@@ -136,10 +138,12 @@ <h3>Interface Definition</h3>
     sequence&lt;ArrayBuffer&gt; getRemoteCertificates ();
     void                  start (RTCQuicParameters remoteParameters);
     void                  stop ();
-    RTCQuicStream         createStream (optional RTCQuicStreamParameters parameters);
+    RTCQuicBidirectionalStream         createBidirectionalStream ();
+    RTCQuicSendStream                  createSendStream ();

```

Comment:
If possible, I would [still](https://github.com/w3c/webrtc-quic/issues/50#issuecomment-420076420) prefer these being promises that resolve when the stream is ready to be used unless there is a good reason why this would be unwise.

FWIW, I would suggest the same for a modern data channel API.

Reply:
I'm open minded to doing this either way. I currently wrote it as throwing an error if the RTCQuicTransport isn't connected, because I figured this is easier for a web developer - Don't create streams until you get an event from your RTCQuicTransport that the state has changed to 'connected'. 

Could you explain the advantages to returning a promise with a stream instead?

Reply:
The main advantage is that one doesn't need to wait for a specific state. The application often only wants to know when it can do its thing and this...

```js
const stream = await quicTransport.createSendStream()
// do stuff...
```

is just easier (and IMO more appealing) than

```js
quicTransport.onstatechange = () => {
  if (quicTransport.state === 'connected') {
    // do stuff...
  }
};
```

(Disclaimer: I omitted error handling for both examples. :wink: )

To support this argument with another discussion going on in this PR: It's a *fire once* event, so a `Promise` is ideal.

Reply:
I'm okay making this change - it's more simple :). 

I'd like to hear other's thoughts before I make this change. I'd prefer not to drag out this PR on issues that we can address after it has merged. If there's some consensus from others for this I'm happy to make the change in this PR, otherwise can we move the discussion to the [issue 50](https://github.com/w3c/webrtc-quic/issues/50) and a subsequent PR?

@robin-raymond @pthatcherg @aboba @steveanton WDYT?

Reply:
Sure, we can move the discussion into #50.

Reply:
I'm in favor of merging and the addressing later.

Reply:
Resolving this conversation, and quoted @lgrahl's comment in the issue.